source/scripts/data/monsters.js

```diff
@@ -31,4 +31,206 @@ export default {
         health: 3,
         strength: 1,
     },
+    WHITE_TROLL: {
+        sprite: DATA.SPRITES.MONSTERS.TROLL,
+        color: DATA.COLORS.WHITE,
+        health: 5,
+        strength: 5,
+    },
+    RED_BAT: {
+        sprite: DATA.SPRITES.MONSTERS.BAT,
+        color: DATA.COLORS.RED,
+        health: 1,
+        strength: 1,
+        movement: function () {
+            if (this.position.x < 0) return {x: +1}
+            if (this.position.x >= DATA.FRAME.WIDTH) return {x: -1}
+            if (this.position.y < 0) return {y: +1}
+            if (this.position.y >= DATA.FRAME.HEIGHT) return {y: -1}
+
+            var choices = [
+                {x: -1},
+                {x: +1},
+                {y: -1},
+                {y: +1}
+            ]
+
+            for(var monster of this.game.monsters) {
+                if (monster.position.x == this.position.x - 1 && monster.position.y == this.position.y)
+                    delete choices[choices.indexOf({x: -1})]
+            }
```

Comment:
How about instead of deleting the choices from a pre-populated array, it fills an empty array? No need to seek through the array (via `indexOf`) to find what to delete.